src/components/LocationsList.js

```diff
@@ -72,55 +67,51 @@ class LocationsList extends InjectedComponent<InjectedProps, Props> {
       id: item.id,
     });
 
-  _onRefresh = (): Promise<*> =>
-    this.injectedProps.locationStore.fetchMany({
-      ...this._getBaseQueryOptions(),
-      skip: 0,
-      take: 20,
-    });
+  _renderItem = ({ item }: { item: Row<Location> }): React.Node => {
+    if (item.value.isLoading()) {
+      return <LoadingListItem />;
+    }
 
-  _renderItem = ({ item }: { item: Location }): React.Node => (
-    <ListItem
-      hideChevron
-      item={item}
-      onPress={this._onItemPress}
-      subtitle={item.summary || '-'}
-      title={item.name}
-    />
-  );
+    if (item.value.hasError()) {
+      return <ErrorListItem />;
+    }
+
+    const location = item.value.getValue();
+
+    return (
+      <ListItem
+        hideChevron
+        item={location}
+        onPress={this._onItemPress}
+        subtitle={location.summary || '-'}
```

Comment:
Can you put the `-`  in a constants file. I think this will be reused a lot.  I'd also prefer that you use `ndash` instead of just `dash`